examples/complete/main.tf

```diff
@@ -188,51 +188,72 @@ resource "aws_service_discovery_http_namespace" "this" {
   tags        = local.tags
 }
 
-module "alb_sg" {
-  source  = "terraform-aws-modules/security-group/aws"
-  version = "~> 5.0"
+module "alb" {
+  source  = "terraform-aws-modules/alb/aws"
+  version = "~> 9.0"
 
-  name        = "${local.name}-service"
-  description = "Service security group"
-  vpc_id      = module.vpc.vpc_id
+  name = local.name
 
-  ingress_rules       = ["http-80-tcp"]
-  ingress_cidr_blocks = ["0.0.0.0/0"]
+  load_balancer_type = "application"
 
-  egress_rules       = ["all-all"]
-  egress_cidr_blocks = module.vpc.private_subnets_cidr_blocks
+  vpc_id  = module.vpc.vpc_id
+  subnets = module.vpc.public_subnets
 
-  tags = local.tags
-}
+  # For example only
+  enable_deletion_protection = false
 
-module "alb" {
-  source  = "terraform-aws-modules/alb/aws"
-  version = "~> 8.0"
+  # Security Group
+  security_group_ingress_rules = {
+    all_http = {
+      from_port   = 80
+      to_port     = 80
+      ip_protocol = "tcp"
+      cidr_ipv4   = "0.0.0.0/0"
+    }
+  }
+  security_group_egress_rules = {
+    all = {
+      ip_protocol = "-1"
+      cidr_ipv4   = module.vpc.vpc_cidr_block
+    }
+  }
 
-  name = local.name
+  listeners = {
+    ex-http = {
+      port     = 80
+      protocol = "HTTP"
 
-  load_balancer_type = "application"
+      forward = {
+        target_group_key = "ex-ecs"
+      }
+    }
+  }
+
+  target_groups = {
+    ex-ecs = {
```

Comment:
Let's not use `-` in keys. I thought `tflint` complains about such already.

Reply:
got it! Updated in https://github.com/terraform-aws-modules/terraform-aws-ecs/pull/130/commits/c69ba10adb76fe1f95d15054a613014204ad6b4c